### DISCUSSION TITLE
Bipartite graph's edge orientation

### DISCUSSION THREAD

**Discussion by zcaicaros** [pyg-team/pytorch_geometric#3708] [Q&A] [answered]
Hi,

Suppose I have a bipartite graph `G` whose adjacency matrix is as follows,
```
adj = [[0, 0, 1],
       [1, 1, 0]]
```
If I am correct, `adj[0, 2] = 1` means `node 0` from part 1 of `G` pointing to `node 2` in the other part of `G`. So here, it makes an implicit assumption that the `NxM` matrix can only model a single orientation. What if I want to model `node 2` in the other part of `G` also points to  `node 0` from part 1 of `G`? I have to create another `adj` like the following.
```
adj_flip = [[0, 1],
            [0, 1],
            [1, 0]]
```
So, for an undirected bipartite graph, do we need two small graphs with different adjacency matrices to describe it?

**Comment by rusty1s** [comment, marked as answer]
Yes, you will need to use two adjacency matrices. Luckily, the reverse adjacency matrix is just the transpose:
```python
edge_index_rev = edge_index.flip([0])
```
